### PR TITLE
Gateway bug fixes

### PR DESCRIPTION
**Pull request by HJianBo**
Some bug fixes for 5.0 beta.3
- Add `endpoint_name`, `like_endpoint_name`, `gte_lifetime`, `lte_lifetime` query parameters for Swagger docs
- More test cases for emqx_gateway_cli
- Optimize the atomicity of the gateway creation procedure and reduce the side effects if creation failed

### CODE REVIEW

File: apps/emqx_gateway/test/emqx_gateway_cli_SUITE.erl [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    emqx_gateway_cli:gateway(["list"]),
+    ?assertEqual(
+      "Gateway(name=coap, status=unloaded)\n"
+      "Gateway(name=exproto, status=unloaded)\n"
+      "Gateway(name=lwm2m, status=unloaded)\n"
+      "Gateway(name=mqttsn, status=unloaded)\n"
+      "Gateway(name=stomp, status=unloaded)\n"
+      , acc_print()).
+
+t_gateway_load(_) ->
```
Comment (HJianBo) [inline apps/emqx_gateway/test/emqx_gateway_cli_SUITE.erl:112]: These following test-cases implementation here will need to delay until the next PR

File: apps/emqx_gateway/src/exproto/emqx_exproto_impl.erl [hunk 2]
```diff
@@ -19,6 +19,14 @@
 
 -behaviour(emqx_gateway_impl).
 
+-include_lib("emqx/include/logger.hrl").
```
Comment (zmstone) [inline apps/emqx_gateway/src/exproto/emqx_exproto_impl.erl:22]: nit: include after export

**Review by zmstone — APPROVED** [review]

File: apps/emqx_gateway/src/mqttsn/emqx_sn_impl.erl [hunk 3]
```diff
@@ -19,6 +19,14 @@
 
 -behaviour(emqx_gateway_impl).
 
+-include_lib("emqx/include/logger.hrl").
+
+-import(emqx_gateway_utils,
```
Comment (zmstone) [inline apps/emqx_gateway/src/mqttsn/emqx_sn_impl.erl:24]: nit: don't import

**Review by k32 — APPROVED** [review]

File: apps/emqx_gateway/src/mqttsn/emqx_sn_impl.erl (hunk 3 above)
Comment (HJianBo) [reply apps/emqx_gateway/src/mqttsn/emqx_sn_impl.erl:24]: Copy that. I will fix them in the next PR

**Pull request merged**